File: src/ServiceControl/Recoverability/Grouping/Api/GroupOperation.cs
using System;

public class GroupOperation
{
    public string Id { get; set; }
    public string Title { get; set; }
    public string Type { get; set; }
    public int Count { get; set; }
    public DateTime? First { get; set; }
    public DateTime? Last { get; set; }
    public string OperationStatus { get; set; }
    public bool? OperationFailed { get; set; }
    public double OperationProgress { get; set; }
    public int? OperationRemainingCount { get; set; }
    public DateTime? OperationStartTime { get; set; }
    public DateTime? OperationCompletionTime { get; set; }
    public bool NeedUserAcknowledgement { get; set; }
}

File: src/resources/GroupOperation.ts
export default interface GroupOperation {
  id: string;
  title: string;
  type: string;
  count: number;
  operation_messages_completed_count?: number;
  comment: string;
  first?: string;
  last?: string;
  operation_status: string;
  operation_failed?: boolean;
  operation_progress: number;
  operation_remaining_count?: number;
  operation_startTime?: string;
  operation_completion_time?: string;
  need_user_acknowledgement: boolean;
  last_operation_completion_time?: string;
}
